frontend/test/pytest/test_preprocess.py

```diff
@@ -147,12 +111,50 @@ def decomposition(self):
         return [qml.RX(*self.parameters, self.wires)]
 
 
+class CustomDevice(Device):
+    """A dummy device from the device API."""
+
+    config = CONFIG_CUSTOM_DEVICE
+
+    def __init__(self, wires, shots=1024):
+        print(pathlib.Path(__file__).parent.parent.parent.parent)
+        super().__init__(wires=wires, shots=shots)
+        dummy_capabilities = get_device_capabilities(self)
+        dummy_capabilities.native_ops.pop("BlockEncode")
+        dummy_capabilities.to_matrix_ops["BlockEncode"] = OperationProperties(False, False, False)
+        self.qjit_capabilities = dummy_capabilities
+
+    @staticmethod
+    def get_c_interface():
+        """Returns a tuple consisting of the device name, and
+        the location to the shared object with the C/C++ device implementation.
+        """
+        system_extension = ".dylib" if platform.system() == "Darwin" else ".so"
+        lib_path = (
+            get_lib_path("runtime", "RUNTIME_LIB_DIR") + "/librtd_null_qubit" + system_extension
+        )
+        return "dummy.remote", lib_path
+
+    def execute(self, circuits, execution_config):
+        """Execution."""
+        return circuits, execution_config
```

Comment:
This implementation doesn't make any sense
```suggestion
    def execute(self, circuits, execution_config):
        """Execution."""
        raise NotImplementedError
```